prometheus-exporters/statsd-exporter/templates/deployment.yaml

```diff
@@ -1,82 +1,76 @@
+{{- range $i, $exporter := .Values.exporters -}}
+{{- if $exporter.enabled }}
+
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: {{ include "statsd-exporter.fullname" . }}
+  name: {{ include "statsd-exporter.fullName" $exporter.name }}
+  namespace: {{ required ".Values.namespace variable missing" $.Values.nameSpace }}
   labels:
-    {{- include "statsd-exporter.labels" . | nindent 4 }}
+    {{- include "statsd-exporter.labels" $exporter.name | nindent 4 }}
 spec:
-  replicas: {{ .Values.replicaCount }}
+  replicas: {{ $.Values.replicaCount }}
+  namespaceSelector:
+    matchNames:
+      - {{ $.Values.nameSpace }}
   selector:
     matchLabels:
-      {{- include "statsd-exporter.selectorLabels" . | nindent 6 }}
+      {{- include "statsd-exporter.labels" $exporter.name | nindent 6 }}
   template:
     metadata:
       labels:
-        {{- include "statsd-exporter.selectorLabels" . | nindent 8 }}
+       {{- include "statsd-exporter.labels" $exporter.name | nindent 8 }}
       annotations:
-        checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") $ | sha256sum }}
+        checksum/config: {{ include (print $.Template.BasePath "/config.yaml") $ | sha256sum }}
```

Comment:
shoudn't this remain configmap.yaml?

Reply:
I have deleted the wrong config file. Just fixed.